### ISSUE TITLE
Precompute some statistics about the graph

### ISSUE THREAD

**Issue by alexkreidler** [oxigraph/oxigraph#94]
I first just want to say thanks for your work! I've now loaded a dataset with 36 million triples into Oxigraph:
![image](https://user-images.githubusercontent.com/11166947/114291630-03b5a400-9a57-11eb-9870-e0f6a690216f.png)

I was wondering if Oxigraph could (perhaps while loading) keep some statistics about the number of triples in the entire graph and each named graph.

This might help queries like the one shown return faster.

**Comment by Tpt** [comment]
Yes, Oxigraph should definitely keep some statistics, at least in order to be able to better optimize query evaluation.

When it is done we might add a piece of code that looks for "count the number of triples" queries and use the statistics for that.
However it is hard to do such query matching properly and using statistics would require them to be exact and not only an approximation.
